File: CS 1600/in class work/person/Person.cpp
//CS 1600
//Person class
#include <string>
#include<iostream>

using namespace std;

class Person
{
	friend ostream& operator <<(ostream&, const Person&);

	private:
		string name;
		int age;
		
	public:
		//default constructor
		Person();
		//Person(string = "", int = 0);
		//Person(string, int):name(n), age(a){}
		//parameterized constructor
		Person( string n, int a );
		
		//function prototype set methods
		void setName(string n);
		void setAge(int a);
		
		//get methods
		string getName()
		{
			return name;
		}
		int getAge()
		{
			return age;
		}
		
		bool operator ==(Person&);
		
		//function prototype utiltiy methods
		void showPerson();
		
	
};

Person::Person()
{
	name = "";
	age = 0;
}

Person::Person(string n, int a)
{
	name = n;
	age = a;
}

//set methods
void Person::setName(string n)
{
	name = n;
}
void Person::setAge(int a)
{
	age = a;
}
/*		
//get methods
string Person::getName()
{
	return name;
}
int Person::getAge()
{
	return age;
}
*/
//utiltiy method
void Person::showPerson()
{
	cout << name << " is " << age << " years old" << endl;
}

bool Person::operator ==(Person &p)
{ 
	return name == p.name && age == p.age;
}

ostream& operator <<(ostream& out, const Person& frnd)
{
	out << frnd.name << " is " << frnd.age << " years old." << endl;
	return out;
}

File: cs 2420/assignments/programming projects/Project 8/Person.java
/*********************************************************** 
 * Programming Assignment 8
 * Recursive method program
 * Programmer: Mark Eatough
 * Course: CS 2410 
 * Created:		March 13
 * Modified:	March 25, 2013 
 * Modified by Mark Eatough
 ***********************************************************/

//Person class
//CS1410, CS2420
//@Override equals(), hashCode(), toString()
//implements Comparable


//package person;

public class Person implements Comparable<Person>
{
	//fields, variables, data members
	private String name;
	private Integer age;
	
	//constructor
	public Person()
	{	
	}

	public Person( String n, Integer a )
	{
		name = n;
		age = a;
	}//end Person()
	
	//accessor methods
	public String getName()
	{
		return name;
	}//end getName()
	
	public Integer getAge()
	{
		return age;
	}//end getAge()
	
	//mutator methods
	public void setName( String n )
	{
		name = n;
	}//end sestName()
	
	public void setAge( Integer a )
	{
		age = a;
	}//end setAge()

	@Override
	public boolean equals( Object o )
	{
		Person p = (Person)o;
		return name.equals(p.name) && age == p.age;
	}//end equals()
	
	@Override
	public int hashCode()
	{
		return name.hashCode() + age.hashCode() * 227;
	}
	
	public int compareTo( Person p )
	{
		return age.compareTo(p.age);
	}
	
	@Override	
	public String toString()
	{
		return getName() + ", " + getAge();
	}//end toString()
	
}//end class Person